src/Components/CategoryFacet.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import { headlessEngine } from "../Engine";
+import {
+  CategoryFacet as CategoryFacetType,
+  CategoryFacetState,
+  buildCategoryFacet,
+  CategoryFacetValue
+} from "@coveo/headless";
+
+// import { makeStyles } from '@material-ui/core/styles';
+import Typography from "@material-ui/core/Typography";
+import IconButton from "@material-ui/core/IconButton";
+import FormLabel from "@material-ui/core/FormLabel";
+import Box from "@material-ui/core/Box";
+import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const hoveredStyle = {
+  cursor: "pointer",
+  color: "primary"
+};
+
+interface ICategoryFacetProps {
+  title: string;
+  field: string;
+  subtitle?: string;
+}
+
+export default class CategoryFacet extends React.Component<
+  ICategoryFacetProps,
+  {}
+> {
+  public static defaultProps = {
+    subtitle: "All Categories"
+  };
+
+  private headlessCategoryFacet: CategoryFacetType;
+  state: CategoryFacetState & {
+    inputValue: "";
+  };
+
+  constructor(props: any) {
+    super(props);
+
+    this.headlessCategoryFacet = buildCategoryFacet(headlessEngine, {
+      options: {
+        numberOfValues: 3,
+        field: this.props.field,
+        delimitingCharacter: "|"
+      }
+    });
+
+    this.state = this.headlessCategoryFacet.state;
+  }
+
+  componentDidMount() {
+    this.headlessCategoryFacet.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.headlessCategoryFacet.subscribe(() => {});
+  }
+
+  updateState() {
+    this.setState(this.headlessCategoryFacet.state);
+  }
+
+  toggleSelect(value: CategoryFacetValue) {
+    this.headlessCategoryFacet.toggleSelect(value);
+  }
+
+  showMore() {
+    this.headlessCategoryFacet.showMoreValues();
+  }
+
+  showLess() {
+    this.headlessCategoryFacet.showLessValues();
+  }
+  //sends left position offset value to getValues <Box> depending on state
```

Comment:
This is _super_ nitpicky, just because the whole point of this example is that people will look through the code, so our comments aren't just for us
```suggestion

  // Calculates the left position offset for values depending on if there are any parent values.
```